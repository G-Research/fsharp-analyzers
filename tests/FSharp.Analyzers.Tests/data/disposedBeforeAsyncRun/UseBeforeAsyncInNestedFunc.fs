module M

open System

let f () =
    
    let asyncReturningFunc () =
        // should warn
        use t1 = new IO.FileStream("x", IO.FileMode.Open)
        // should warn
        use t2 = new IO.FileStream("x", IO.FileMode.Open)
        async {
            // should not warn
            use t3 = new IO.FileStream("x", IO.FileMode.Open)
            return 23
    }

    asyncReturningFunc ()
