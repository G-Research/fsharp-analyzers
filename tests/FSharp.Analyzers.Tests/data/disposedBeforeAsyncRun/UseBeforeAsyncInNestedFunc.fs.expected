GRA-DISPBEFOREASYNC-001 | Warning | (9,12 - 9,14) | Object is disposed before returned async is run | []
GRA-DISPBEFOREASYNC-001 | Warning | (11,12 - 11,14) | Object is disposed before returned async is run | []
